pkg/kapp/preflight/check.go

```diff
@@ -36,6 +38,10 @@ func (cf *checkImpl) SetEnabled(enabled bool) {
 	cf.enabled = enabled
 }
 
+func (cf *checkImpl) SetConfig(_ map[string]any) error {
+	return nil
+}
```

Comment:
For this, maybe we should update the `checkImpl` and `CheckFunc` types to have the ability for configuration? The `CheckFunc` type could be updated to take it is as a parameter, with the implementation ignoring it if they don't need it. The `checkImpl` type could be updated to have a `config map[string]any` field it passes to the `checkFunc` when it calls it.

Could totally be done in a follow up, but just something I thought might be worth considering.

Reply:
That would be testing, which I haven't included here; as I wanted to get the basic functionality written before spending additional time on the testing.

Reply:
>That would be testing, which I haven't included here

Not sure I'm following, how is this testing specific?

Reply:
The `checkImpl` and `CheckFunc` are used primarily for testing; it's not an actual PreflightCheck.

i.e. the references to `NewCheck` which creates those types are only in registry_test.c

Basically, this PR is just the implementation without any test code.

Reply:
It can be used to implement an actual preflight check even though it is currently only used in the tests. OK with this being done in a future PR though.